esphome/components/atm90e26/atm90e26.cpp

```diff
@@ -0,0 +1,235 @@
+#include "atm90e26.h"
+#include "atm90e26_reg.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace atm90e26 {
+
+static const char *const TAG = "atm90e26";
+
+void ATM90E26Component::update() {
+  if (this->read16_(ATM90E26_REGISTER_FUNCEN) != 0x0030) {
+    this->status_set_warning();
+    return;
+  }
+
+  if (this->voltage_sensor_ != nullptr) {
+    this->voltage_sensor_->publish_state(this->get_line_voltage_());
+  }
+  if (this->current_sensor_ != nullptr) {
+    this->current_sensor_->publish_state(this->get_line_current_());
+  }
+  if (this->power_sensor_ != nullptr) {
+    this->power_sensor_->publish_state(this->get_active_power_());
+  }
+  if (this->reactive_power_sensor_ != nullptr) {
+    this->reactive_power_sensor_->publish_state(this->get_reactive_power_());
+  }
+  if (this->power_factor_sensor_ != nullptr) {
+    this->power_factor_sensor_->publish_state(this->get_power_factor_());
+  }
+  if (this->forward_active_energy_sensor_ != nullptr) {
+    this->forward_active_energy_sensor_->publish_state(this->get_forward_active_energy_());
+  }
+  if (this->reverse_active_energy_sensor_ != nullptr) {
+    this->reverse_active_energy_sensor_->publish_state(this->get_reverse_active_energy_());
+  }
+  if (this->freq_sensor_ != nullptr) {
+    this->freq_sensor_->publish_state(this->get_frequency_());
+  }
+  this->status_clear_warning();
+}
+
+void ATM90E26Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up ATM90E26 Component...");
+  this->spi_setup();
+
+  uint16_t mmode = 0x422;  // default values for everything but L/N line current gains
+  mmode |= l_line_gain_ << 13;
+  mmode |= n_line_gain_ << 11;
```

Comment:
This is currently overwriting the right-most bit of L line gain.

```suggestion
  mmode |= n_line_gain_ << 10;
```